lib/api/bucketPut.js

```diff
@@ -41,7 +41,7 @@ export default function bucketPut(authInfo, request, locationConstraint, log,
     }
     // - AWS JS SDK sends a request with locationConstraint us-east-1
     // if no locationConstraint provided.
-    if (locationConstraint && config.locationConstraints &&
+    if (locationConstraint && !config.regions && config.locationConstraints &&
```

Comment:
If we have both the legacy property of config.regions **and** the new property of locationConstraint, we should throw an error when parsing the config and the server shouldn't start.  Same thing if we have locationConstraints but are missing restEndpoints.